src/main/java/net/dries007/tfc/common/blockentities/rotation/WindmillBlockEntity.java

```diff
@@ -220,4 +227,31 @@ public Node getRotationNode()
     {
         return node;
     }
+
+    public boolean hasFullIdenticalSet()
+    {
+        ItemStack stack = inventory.getStackInSlot(0);
+
+        // catch anything that's not a windmill blade
+        for (int i = 1; i < SLOTS; i++)
+        {
+            if (!inventory.getStackInSlot(i).is(TFCTags.Items.ALL_WINDMILL_BLADES))
+            {
+                return false;
+            }
+        }
+
+        WindmillBladeItem item = (WindmillBladeItem) stack.getItem();
+        WindmillBladeItem.BladeModel model = item.getModel();
+
+        for (int i = 1; i < SLOTS; i++)
+        {
+            if (((WindmillBladeItem) inventory.getStackInSlot(i).getItem()).getModel() != model)
+            {
+                return false;
+            }
+        }
```

Comment:
As alluded to above, both of these casts should be done defensively, i.e. `if (stack.getItem() instanceof WindmillBladeItem item)`. The prior tag check doesn't prevent these from erroring.